### PR TITLE
Fixing minitest 5.10 and misc deprecations

### LINKED ISSUE #219: error with Minitest 5.10
I get the following error:

    # Running tests with run options --seed 31173:

    /Users/bruce/.rvm/gems/ruby-2.3.3@havn/gems/minitest-5.10.0/lib/minitest.rb:667:in `block in prerecord': undefined method `prerecord' for #<Minitest::Reporters::DelegateReporter:0x007f86811c5250> (NoMethodError)
	from /Users/bruce/.rvm/gems/ruby-2.3.3@havn/gems/minitest-5.10.0/lib/minitest.rb:666:in `each'
	from /Users/bruce/.rvm/gems/ruby-2.3.3@havn/gems/minitest-5.10.0/lib/minitest.rb:666:in `prerecord'
	from /Users/bruce/.rvm/gems/ruby-2.3.3@havn/gems/minitest-5.10.0/lib/minitest.rb:322:in `run_one_method'
	from /Users/bruce/.rvm/gems/ruby-2.3.3@havn/gems/minitest-5.10.0/lib/minitest.rb:310:in `block (2 levels) in run'
	from /Users/bruce/.rvm/gems/ruby-2.3.3@havn/gems/minitest-5.10.0/lib/minitest.rb:309:in `each'
	from /Users/bruce/.rvm/gems/ruby-2.3.3@havn/gems/minitest-5.10.0/lib/minitest.rb:309:in `block in run'
	from /Users/bruce/.rvm/gems/ruby-2.3.3@havn/gems/minitest-5.10.0/lib/minitest.rb:349:in `on_signal'
	from /Users/bruce/.rvm/gems/ruby-2.3.3@havn/gems/minitest-5.10.0/lib/minitest.rb:336:in `with_info_handler'
	from /Users/bruce/.rvm/gems/ruby-2.3.3@havn/gems/minitest-5.10.0/lib/minitest.rb:308:in `run'
	from /Users/bruce/.rvm/gems/ruby-2.3.3@havn/gems/minitest-5.10.0/lib/minitest.rb:158:in `block in __run'
	from /Users/bruce/.rvm/gems/ruby-2.3.3@havn/gems/minitest-5.10.0/lib/minitest.rb:158:in `map'
	from /Users/bruce/.rvm/gems/ruby-2.3.3@havn/gems/minitest-5.10.0/lib/minitest.rb:158:in `__run'
	from /Users/bruce/.rvm/gems/ruby-2.3.3@havn/gems/minitest-5.10.0/lib/minitest.rb:135:in `run'
	from /Users/bruce/.rvm/gems/ruby-2.3.3@havn/gems/minitest-5.10.0/lib/minitest.rb:62:in `block in autorun'

all I am doing is this in `testhelper.rb`:

    Minitest::Reporters.use! Minitest::Reporters::MeanTimeReporter.new

I'm guessing `DelegateReporter` should just inherit from `Minitest::Reporter` (or a subclass - instead of just implementing prerecord) to also prevent these issues in the future, but I haven't dug deep enough to be sure...

### PR DESCRIPTION
**Pull request by brucek**
This should fix #219 and a few deprecations that came up

### CODE REVIEW

File: test/unit/minitest/spec_reporter_test.rb [hunk 1]
```diff
@@ -10,15 +10,15 @@ def setup
 
     def test_removes_underscore_in_name_if_shoulda
       @test.name = "test_: Should foo"
-      assert_output /test:/ do
+      assert_output(/test:/) do
```
Comment (zenspider) [inline test/unit/minitest/spec_reporter_test.rb]: Not my project... but refrain from style changes in PRs, esp bug fix PRs

**Comment by zenspider** [comment]
I can also put out `5.10.1` that does a `respond_to?` check today... if that helps smooth things out.

**Comment by os97673** [comment]
Thank you for the PR.  I've applied subclassing part.  As for style changes: if you feel that it is important please file a separate PR. (I agree with @zenspider that style change should not be mixed with bug fixes)

**Comment by brucek** [comment]
OK - that's fine, although it seems to me that fixing deprecation warnings would be a bit more than just a "style change" and since the goal (IMO) of this issue (since I submitted it) was resolving everything having to do with updating to minitest 5.10, it seemed OK to include here.

Just trying to understand the culture.  Anyways, new issue in #221 and fix in #222.

**Comment by os97673** [comment]
> Just trying to understand the culture.

No problems, there are two things:
1. I do not like commits which contain independent changes
2. #219 says about missing methods and nothing about deprecated warnings and this PR too.

BTW you do not have to file issue to file PR, you can just file PR with description of the problem you tries to fix. (and thank you for your PRs)

**Comment by brucek** [comment]
Copy that.  Just to make sure I got you right (sorry, not trying to belabor the point):

1.  Separate out commits to be more semantically independent (ie bugfix vs deprecation fixes), don't worry about having a higher commit count
2.  I might misunderstand slightly, but the title of this PR *is* "Fixing minitest 5.10 **and misc deprecations**"...  I do see your point that the deprecations weren't mentioned in #219 though, nor did I document the actual deprecation/warning messages in either place.

Thank you for helping me figure out how to be a better contributor (and for all your work on the gem) - I had also generally been assuming I needed an issue for every PR, so very nice to know that submitting naked PRs is an accepted practice too.

**Comment by os97673** [comment]
> Separate out commits to be more semantically independent (ie bugfix vs deprecation fixes), don't worry about having a higher commit count

yes, I'm not charged for number of commits the repo contains ;)

> I might misunderstand slightly, but the title of this PR is "Fixing minitest 5.10 and misc deprecations"... I do see your point that the deprecations weren't mentioned in #219 though, nor did I document the actual deprecation/warning messages in either place.

Well,  it is kind of my interpretation of the text :)  I've seen issue which described problem with messing method, I see the PR which fixes the problem and "misc deprecations".  So, for me the main problem is missed method and it is not clear why I should fix deprecations perhaps couple lines in PR would clarify your position (something like: "I've also noticed that tests use deprecated API and I've fixed this" - you've spent few second to explain the problem for me and it is more clear/important for me ;) .  
But as I said the main reason why I've not merged the PR as is was the mix of bug fixe and cleanup in one commit, and I was too lazy to apply optional changes manually :(

Hope this clarify the situation for you.  But feel free to ask more questions if you have them. (btw there is a gitter chat you can use for such discussions)

**Pull request closed without merging**